src/dbnode/storage/shard.go

```diff
@@ -1964,7 +1959,7 @@ func (s *dbShard) ColdFlush(
 			// Cold flushes can only happen on blockStarts that have been
 			// warm flushed, because warm flush logic does not currently
 			// perform any merging logic.
-			if !s.hasWarmFlushed(t.ToTime()) {
+			if !s.IsBlockRetrievable(t.ToTime()) {
```

Comment:
Can you actually keep the duplicated method `hasWarmFlushed`? I know its lame to duplicate but I can easily see someone changing the meaning of `IsBlockRetrievable` and not expecting the impact it'll have on this

Reply:
Sure.